### ISSUE TITLE
Find replacement for deprecated `ProgressDialog`

### ISSUE THREAD

**Issue by wmontwe** [k9mail/k-9#7696]
With API 26 the use of `ProgressDialog` is [deprecated](https://developer.android.com/reference/android/app/ProgressDialog). 

> ProgressDialog is a modal dialog, which prevents the user from interacting with the app. Instead of using this class, you should use a progress indicator like ProgressBar, which can be embedded in your app's UI. Alternatively, you can use a notification to inform the user of the task's progress.

We need to find a replacement for the various places the `ProgressDialog` is used.

**Comment by MonuGit9** [comment]
Assign me

**Comment by cketti** [comment]
@MonuGit9: We probably want to replace progress dialogs with a non-modal mechanism to show the progress. This requires coming up with a new UI design (hence the `status: needs design` label). 

If you have a concrete suggestion for a specific case where a `ProgressDialog` is currently used, please open a new issue with your suggested solution (and reference this issue). Someone from our design team will have a look and sign off on your design or suggest changes. Please don't start with an implementation before a design has been approved.
